rmw_connext_shared_cpp/src/shared_functions.cpp

```diff
@@ -319,6 +323,10 @@ create_node(const char * implementation_identifier, const char * name, size_t do
   }
   participant_qos.participant_name.name =
     static_cast<char *>(rmw_allocate((strlen(name) + 1) * sizeof(char)));
+  if (!participant_qos.participant_name.name) {
+    RMW_SET_ERROR_MSG("failed to allocate memory");
+    return NULL;
+  }
   participant_qos.participant_name.name = strdup(name);
```

Comment:
I had to remove the extra free. DDS appears to free this for us, because I get this on exit if I leave the free in:

```
[D0087|DELETE Sub]RTIOsapiHeap_freeBufferAligned:inconsistent free/alloc: block id 0X1 being freed with "RTIOsapiHeap_freeBufferAligned" and was allocated with "RTIOsapiHeap_unknownFunction"
talker(63898,0x7fff72c12000) malloc: *** error for object 0x7fc288c17cc0: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
```

See: 44fb031